web_src/js/components/RepoActionView.vue

```diff
@@ -0,0 +1,514 @@
+<template>
+  <div class="action-view-container">
+    <div class="action-view-header">
+      <div class="action-info-summary">
+        {{ runInfo.title }}
+      </div>
+    </div>
+    <div class="action-view-body">
+      <div class="action-view-left">
+        <div class="job-group-section" v-for="(jobGroup, i) in allJobGroups" :key="i">
+          <!--        <div class="job-group-summary">-->
+          <!--          {{ jobGroup.summary }}-->
+          <!--        </div>-->
+          <div class="job-brief-list">
+            <a class="job-brief-item" v-for="(job, index) in jobGroup.jobs" :key="job.id" :href="runInfo.htmlurl+'/jobs/'+index">
+              <SvgIcon name="octicon-check-circle-fill" class="green" v-if="job.status === 'success'"/>
+              <SvgIcon name="octicon-skip" class="ui text grey" v-else-if="job.status === 'skipped'"/>
+              <SvgIcon name="octicon-clock" class="ui text yellow" v-else-if="job.status === 'waiting'"/>
+              <SvgIcon name="octicon-blocked" class="ui text yellow" v-else-if="job.status === 'blocked'"/>
+              <SvgIcon name="octicon-meter" class="ui text yellow" class-name="job-status-rotate" v-else-if="job.status === 'running'"/>
+              <SvgIcon name="octicon-x-circle-fill" class="red" v-else/>
+              {{ job.name }}
+              <button class="job-brief-rerun" @click="rerunJob(index)" v-if="job.can_rerun">
```

Comment:
`v-if="job.can_rerun" && !job.is_successful`.
Either that way, or adapt `job.can_rerun`.
I'm fine with both approaches, but displaying the sync icon readonly is a bad idea.
Or why else does nothing happen when I click on it when when the job was already successful?

Reply:
Also, this looks like a wrong cursor for it: 
![image](https://user-images.githubusercontent.com/51889757/210333883-4db8bce7-b8cd-497b-a3cf-f536ed902d8d.png)
Doesn't look like a button from the cursor…

Reply:
> `v-if="job.can_rerun" && !job.is_successful`.

I don't think so, we can also rerun a successful step.

> Doesn't look like a button from the cursor…

It should be better now:

![2023-01-12 16 52 33](https://user-images.githubusercontent.com/9418365/212022425-d47f853e-c008-4c4e-b2f0-9ca148a9d3d9.gif)

Reply:
Yeah, I think the problem was the sqlite deadlock, combined with the wrong cursor which led me to believe that something is wrong.